fedimint-core/src/db/mod.rs

```diff
@@ -761,9 +866,10 @@ impl<'parent> DatabaseTransaction<'parent> {
     pub fn new(
         dbtx: Box<dyn ISingleUseDatabaseTransaction<'parent>>,
         decoders: ModuleDecoderRegistry,
+        notifications: &'parent Notifications,
     ) -> DatabaseTransaction<'parent> {
         DatabaseTransaction {
-            tx: dbtx,
+            tx: Box::new(NotifyingTransactionWrapper::new(dbtx, notifications)),
```

Comment:
With all these wrappers having some architecture diagram in the docs would be very useful. Maybe you can work together with @m1sterc001guy on that?

It's becoming increasingly hard to keep the full architecture in my mind and up to date, so I imagine it's even worse for new contributors. [Mermaid](https://github.com/mermaid-js/mermaid) in markdown docs works quite well for these kind of things since it's easily diffable.

Reply:
+1 Created an issue: https://github.com/fedimint/fedimint/issues/1831